internal/services/cragcast/cragcast.go

```diff
@@ -0,0 +1,227 @@
+
+package cragcast
+
+import (
+	"errors"
+	"sort"
+	"strings"
+	"time"
+
+	client "github.com/dsauerbrun/cragcast/pkg/weather-client"
+	noaaTypes "github.com/dsauerbrun/cragcast/pkg/weather-client/noaa"
+)
+
+func GetForecast(cragId int) (*Cragcast, error) {
+	// TODO: get crag's lat/lng from DB based on cragId. For now, we use boulder which is 40.0294122,-105.3223779
+	lat, lng := 40.0294122, -105.3223779
+	weatherClientForecast, err := client.New().GetForecast(lat, lng)
+	if err != nil {
+		return nil, err
+	}
+
+	// what do people actually care about in a climbing forecast?
+	// rainfall %, amount of rainfall
+	// temps by the hour as well as the highest temp
+	// cloudiness
+	// humidity by the hour as well as highest humidity
+	// TODO: People typically care about the duration from morning until night so we should adjust for location's timezone and only take into account daylight hours
+
+	hourlyForecasts, err := getProcessedHours(weatherClientForecast)
+	if err != nil {
+		return nil, err
+	}
+	cragcast := &Cragcast{
+		Days: []*DailyForecast{},
+	}
+
+	for _, hour := range hourlyForecasts {
+		cragcastDayIndex := len(cragcast.Days) - 1
+		// do we need a timezone conversion here?
+		day := hour.StartOfHour.Day()
+		if len(cragcast.Days) == 0 || cragcast.Days[cragcastDayIndex].Date.Day() != day {
+			// we are on a new day so we append a day to our cragcast
+			initialLowFahrenheit := float32(1000)
+			cragcast.Days = append(cragcast.Days, &DailyForecast{
+				Date:               hour.StartOfHour,
+				LowOfDayFahrenheit: &initialLowFahrenheit,
+				Hours:              []*HourlyForecast{},
+			})
+			cragcastDayIndex++
+		}
+
+		cragcastDay := cragcast.Days[cragcastDayIndex]
+		if hour.TotalPrecipMm != nil {
+			newPrecipMm := getPointerValueWithDefault(cragcastDay.TotalPrecipMm) + *hour.TotalPrecipMm
+			cragcastDay.TotalPrecipMm = &newPrecipMm
+		}
+
+		if hour.ChanceOfPrecip != nil && getPointerValueWithDefault(cragcastDay.ChanceOfPrecip) < *hour.ChanceOfPrecip {
+			cragcastDay.ChanceOfPrecip = hour.ChanceOfPrecip
+		}
+		if hour.TemperatureFahrenheit != nil && getPointerValueWithDefault(cragcastDay.LowOfDayFahrenheit) > *hour.TemperatureFahrenheit {
+			cragcastDay.LowOfDayFahrenheit = hour.TemperatureFahrenheit
+		}
+		if hour.TemperatureFahrenheit != nil && getPointerValueWithDefault(cragcastDay.HighOfDayFahrenheit) < *hour.TemperatureFahrenheit {
+			cragcastDay.HighOfDayFahrenheit = hour.TemperatureFahrenheit
+		}
+
+		if hour.Humidity != nil && getPointerValueWithDefault(cragcastDay.HumidityHigh) < *hour.Humidity {
+			cragcastDay.HumidityHigh = hour.Humidity
+		}
+
+		if hour.snowfallMm != nil {
+			newSnowfall := getPointerValueWithDefault(cragcastDay.snowfallMm) + *hour.snowfallMm
+			cragcastDay.snowfallMm = &newSnowfall
+		}
+
+		cragcastDay.Hours = append(cragcastDay.Hours, hour)
+	}
+
+	return cragcast, nil
+}
+
+func getProcessedHours(weatherClientForecast *noaaTypes.GridpointGeoJson) ([]*HourlyForecast, error) {
```

Comment:
As i'm reviewing this function, it's hard for me to reason about what its purpose is at a surface level. Likely because it's called get**Processed**Hours, where I'm left to reason about what the "processing" is. Perhaps a rename and, if needed after the rename, some comments around intent?